test/run_test.py

```diff
@@ -904,7 +906,7 @@ def get_pytest_args(options, is_cpp_test=False):
     else:
         # When under the normal mode, retry a failed test 2 more times. -x means stop at the first
         # failure
-        rerun_options = ["-x", "--reruns=2", "--sw"]
+        rerun_options = ["-x", "--reruns=2", f"--sc={stepcurrent_key}"]
```

Comment:
I think we should limit this behavior to CI only, as I suspect that this is not the behavior I need when running a test file locally.  I would be in for a surprise if some tests are skipped because I have run them before.  On CI, we can assume that the cache directory would be in our control, i.e. it's cleaned up every time at the moment.